packages/botonic-react/src/components/multichannel/multichannel-carousel.jsx

```diff
@@ -1,112 +1,112 @@
+    const { imageProps, title, subtitle, buttons } =
+      extractElementProperties(element)
+
+    const textMessage = getTextMessage(requestContext.session, title, subtitle)
+
+    if (imageProps?.src) {
+      const messageWithImage = <Image src={imageProps.src} />
+      messages.push(messageWithImage)
+    }
+
+    if (
+      isWhatsapp(requestContext.session) &&
+      buttons.some(button => button.url)
+    ) {
+      const messageWithButtonUrl = (
+        <WhatsappCTAUrlButton
+          key={`carousel-element-${i}-cta-url`}
+          body={title}
+          footer={`_${subtitle}_`}
+          displayText={buttons[0].text}
+          url={buttons[0].url}
+        />
+      )
+      messages.push(messageWithButtonUrl)
+    }
+
+    const messageWithButtons = buttons.some(button => button.payload) ? (
+      <Text key={`carousel-element-${i}-text`}>
+        {textMessage}
+        {buttons
+          .filter(button => isWhatsapp(requestContext.session) && !button.url)
+          .map(button => (
+            <Button
+              key={`carousel-element-${i}-button`}
+              payload={button.payload}
+              url={button.url}
+            >
+              {button.text}
+            </Button>
+          ))}
+      </Text>
+    ) : (
+      []
+    )
+    messages.push(messageWithButtons)
+  })
+
+  return <>{messages}</>
+}
+
+function getButtons(node) {
+  return [].concat(getFilteredElements(node, isMultichannelButton))
+}
+
+function extractElementProperties(element) {
+  let imageProps, title, subtitle
+  const buttonsChildren = []
+
+  for (const node of element) {
+    if (isNodePic(node)) imageProps = node.props
+    if (isNodeTitle(node)) title = node.props.children
+    if (isNodeSubtitle(node)) subtitle = node.props.children
+    if (isMultichannelButton(node)) buttonsChildren.push(node)
+    if (Array.isArray(node)) buttonsChildren.push(...getButtons(node))
+  }
+
+  const buttons = buttonsChildren.map(button => {
+    return {
+      text: button.props.children,
+      payload: button.props.payload,
+      url: button.props.url,
+    }
+  })
+
+  return { imageProps, title, subtitle, buttons }
 }
 
-// const carouselToCaption = (index, title, subtitle, imageSrc, buttonProps) => {
-//   let caption = ''
-//   let header = `${title ? `*${title}*` : ''}`
-//   header += `${subtitle ? ` - _${subtitle}_` : ''}`
-//   let buttons = ''
-//   if (buttonProps.url) {
-//     buttons += ` - ${buttonProps.children}: ${buttonProps.url}`
-//     caption = `${header ? `${header}\n` : ''}${buttons}`
-//   }
-//   if (buttonProps.payload || buttonProps.path) {
-//     buttons += `${index}. `
-//     caption = `${buttons}${buttonProps.children}`
-//   }
-//   return caption
-// }
+function getTextMessage(session, title, subtitle) {
+  const textMessage = `**${title}**${subtitle ? ` _${subtitle}_` : ''}`
+  return isWhatsapp(session) || isFacebook(session)
+    ? convertToMarkdownMeta(textMessage)
+    : textMessage
```

Comment:
i would rename this variable as `formattedTextMessage` to be clearer it will be formatted in both cases